src/amazon/dsstne/engine/NNNetwork.cpp

```diff
@@ -95,61 +95,61 @@ ostream& operator<< (ostream& out, NNNetworkDescriptor& d)
 
 bool ValidateNetworkDescriptor(NNNetworkDescriptor& d)
 {
-    
-
-
     return true;
 }
 
-tuple<NNFloat, uint32_t, NNFloat, NNFloat> NNNetwork::GetLRN()
+tuple<NNFloat, uint32_t, NNFloat, NNFloat> NNNetwork::GetLRN() const
 {
     return make_tuple(_LRN_k, _LRN_n, _LRN_alpha, _LRN_beta);
 }
 
-tuple<uint32_t> NNNetwork::GetMaxout()
+tuple<uint32_t> NNNetwork::GetMaxout() const
 {
     return make_tuple(_maxout_k);
 }
 
-tuple<NNFloat, NNFloat> NNNetwork::GetSparsenessPenalty()
+tuple<NNFloat, NNFloat> NNNetwork::GetSparsenessPenalty() const
 {
     return make_tuple(_sparsenessPenalty_p, _sparsenessPenalty_beta);
 }
 
-tuple<NNFloat> NNNetwork::GetDenoising()
+tuple<NNFloat> NNNetwork::GetDenoising() const
 {
     return make_tuple(_denoising_p);
 }
 
-tuple<NNFloat, NNFloat> NNNetwork::GetDeltaBoost()
+tuple<NNFloat, NNFloat> NNNetwork::GetDeltaBoost() const
 {
     return make_tuple(_deltaBoost_one, _deltaBoost_zero);
 }
 
-tuple<NNFloat, NNFloat, NNFloat, NNFloat> NNNetwork::GetSMCE()
+tuple<NNFloat, NNFloat, NNFloat, NNFloat> NNNetwork::GetSMCE() const
 {
     return make_tuple(_SMCE_oneTarget, _SMCE_zeroTarget, _SMCE_oneScale, _SMCE_zeroScale);
 }
 
-tuple<bool> NNNetwork::GetShuffleIndices()
+tuple<bool> NNNetwork::GetShuffleIndices() const
 {
     return make_tuple(_bShuffleIndices);
 }
 
-tuple<string, int32_t> NNNetwork::GetCheckPoint()
+tuple<string, int32_t> NNNetwork::GetCheckPoint() const
 {
     return make_tuple(_checkpoint_name, _checkpoint_interval);
 } 
 
-NNLayer* NNNetwork::GetLayer(const string& layer)
+const NNLayer* NNNetwork::GetLayer(const string& layer) const
 {
-    NNLayer* pLayer         = _mLayer[layer];
-    if (pLayer == NULL)
+    const map<string, NNLayer*>::const_iterator itr = _mLayer.find(layer);
```

Comment:
here you you use: 
const map<string, NNLayer*>::const_iterator itr 
and later you use: 
const auto itr
It is better to be consistent withing one file